app/src/main/java/com/google/maps/android/compose/RecompositionActivity.kt

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.maps.android.compose
+
+import android.os.Bundle
+import android.util.Log
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.Button
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import com.google.android.gms.maps.model.Marker
+import com.google.maps.android.compose.theme.MapsComposeSampleTheme
+import kotlin.random.Random
+
+private const val TAG = "RecompositionActivity"
+
+/**
+ * This is a sample activity showcasing how the recomposition works. The location is changed
+ * every time we click on the button, and the marker gets updated (removed and added in a new
+ * location)
+ */
+class RecompositionActivity : ComponentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            val cameraPositionState = rememberCameraPositionState {
+                position = defaultCameraPosition
+            }
+            Box(Modifier.fillMaxSize()) {
+                MapsComposeSampleTheme {
+                    GoogleMapView(
+                        modifier = Modifier.matchParentSize(),
+                        cameraPositionState = cameraPositionState
+                    )
+                }
+            }
+        }
+    }
+
+    @Composable
+    fun GoogleMapView(
+        modifier: Modifier = Modifier,
+        cameraPositionState: CameraPositionState = rememberCameraPositionState(),
+        content: @Composable () -> Unit = {},
+    ) {
+        var location by remember { mutableStateOf(singapore) }
+        val singaporeState = MarkerState(position = location)
```

Comment:
`MarkerState.position` is a var with a public setter and a `mutableStateOf` behind it, and it is observed for changes by `Marker()` machinery. So it is not necessary to keep a separate remembered `mutableStateOf` to track position changes, or to recreate MarkerState on every recomposition.
Consider this instead:
```
val singaporeState = remember { MarkerState(singapore) }
```
Then update via:

```
singaporeState.position = when (randomValue) {
    0 -> singapore
    1 -> singapore2
    2 -> singapore3
    else -> singapore
```

Reply:
@wangela @kikoso if this pattern does not work then there has been a regression that needs to be fixed. Various recently filed issues and my own testing suggest that this is the case. The pattern used to work flawlessly and is the way the API is intended to be used; it's described at https://github.com/androidx/androidx/blob/androidx-main/compose/docs/compose-api-guidelines.md#hoisted-state-types and subsequent sections.

Recreating MarkerState on every recomposition is akin to the following, which the IDE will tell you is missing 'remember':
```
val state = mutableStateOf(1)
```

Reply:
@wangela @kikoso FYI: the regression (if it is just a single one) was introduced in 2.14.1